Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditor.cpp

```diff
@@ -31,15 +31,91 @@ namespace AzToolsFramework
         }
     }
 
-    DPERowWidget::DPERowWidget(QWidget* parentWidget)
+    QSize DPELayout::sizeHint() const
+    {
+        int cumulativeWidth = 0;
+        int preferredHeight = 0;
+
+        // sizeHint for this horizontal layout is the sum of the preferred widths,
+        // and the maximum of the preferred heights
+        for (int layoutIndex = 0; layoutIndex < count(); ++layoutIndex)
+        {
+            QWidget* widgetChild = itemAt(layoutIndex)->widget();
+            if (widgetChild)
+            {
+                auto widgetSizeHint = widgetChild->sizeHint();
+                cumulativeWidth += widgetSizeHint.width();
+                preferredHeight = qMax(widgetSizeHint.height(), preferredHeight);
+            }
+        }
+        return { cumulativeWidth, preferredHeight };
+    }
+
+    QSize DPELayout::minimumSize() const
+    {
+        int cumulativeWidth = 0;
+        int minimumHeight = 0;
+
+        // minimumSize for this horizontal layout is the sum of the min widths,
+        // and the maximum of the preferred heights
+        for (int layoutIndex = 0; layoutIndex < count(); ++layoutIndex)
+        {
+            QWidget* widgetChild = itemAt(layoutIndex)->widget();
+            if (widgetChild)
+            {
+                const auto minWidth = widgetChild->minimumSizeHint().width();
+                ;
+                if (minWidth > 0)
+                {
+                    cumulativeWidth += minWidth;
+                }
+                minimumHeight = qMax(widgetChild->sizeHint().height(), minimumHeight);
+            }
+        }
+        return { cumulativeWidth, minimumHeight };
+    }
+
+    void DPELayout::setGeometry(const QRect& rect)
+    {
+        QLayout::setGeometry(rect);
+
+        // for now, just divide evenly horizontally. Later, we'll implement QSplitter-like
+        // functionality to allow the user to resize columns within a DPE
```

Comment:
Unless this is a marker for future development, I'm not sure how useful this comment is :)

Reply:
it is, but I made it clearer as a "todo"